src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -168,9 +171,15 @@ void GcsJobManager::HandleGetNextJobID(const rpc::GetNextJobIDRequest &request,
 }
 
 std::string GcsJobManager::GetRayNamespace(const JobID &job_id) const {
-  auto it = ray_namespaces_.find(job_id);
-  RAY_CHECK(it != ray_namespaces_.end()) << "Couldn't find job with id: " << job_id;
-  return it->second;
+  auto it = cached_job_configs_.find(job_id);
+  RAY_CHECK(it != cached_job_configs_.end()) << "Couldn't find job with id: " << job_id;
+  return it->second->ray_namespace();
+}
+
+int32_t GcsJobManager::GetNumJavaWorkersPerProcess(const JobID &job_id) const {
```

Comment:
Nit, I think we only need to expose a `GetJobConfig` method, and use `GetJobConfig().num_java_workers_per_process()` at the call sites. 
Otherwise there will be too many such methods.

Reply:
+1

Reply:
Good point. I'm going to refine this at #21279 because there still be an issue that when we should clear the cache.